### DISCUSSION TITLE
Error reading data with spectral reflectance attributes

### DISCUSSION THREAD

**Discussion by seanreilly66** [r-lidar/lidR#658] [Q&A] [answered]
Hi everyone, 

I've run into a hurdle with which I would really appreciate some help.

I have a collection of las files generated from structure from motion which have additional spectral attributes recorded as decimal reflectance values, rather than 8 bit integers. These loaded fine in the old version of lidR but we have been unable to open them since the major updates a bit ago. As luck would have it, most colors had been added as full names (e.g., red instead of R) so lidR does not flag them as a default attribute. However, our near infrared data is stored as nir. Consequently, we get the following error when attempting to load a file:

Attribute 'nir' renamed 'NIR' to match with default attribute names.
Error: Invalid data: NIR is not an integer

Is there a way around this within readLAS? Alternatively, is there a way to modify the attribute name (e.g., change it to nir_reflectance) so lidR will not flag it? 

Thank you greatly for any help.

Sean

**Comment by Jean-Romain** [comment, marked as answer]
> These loaded fine in the old version of lidR but we have been unable to open them since the major updates a bit ago

Please report an issue with a minimal reproducible example. Is is related to https://github.com/r-lidar/rlas/issues/59 ?

> Alternatively, is there a way to modify the attribute name (e.g., change it to nir_reflectance) so lidR will not flag it?

Sure, use `names()`

```r
library(lidR)
LASfile <- system.file("extdata", "Megaplot.laz", package="lidR")
las = readLAS(LASfile)
n = names(las@data)
n[4] = "time"
names(las@data) = n
names(las)
```

You can tweak everything manually to reach your goal but I need a reproducible example to help

**Reply by seanreilly66** [reply]
Thanks for the help. Here's a simple reprex of the problem.

```r
library(lidR)

las <- LAS(data = data.frame(
  X = runif(100, 1, 10),
  Y = runif(100, 1, 10),
  Z = runif(100, 1, 10)
))

las <- LAS(df)

nir = runif(100)

las <- add_lasattribute(las, x = nir, name = 'nir', desc = 'nir')

writeLAS(las, 'temp/nir_error.las')

readLAS('temp/nir_error.las')
```


I hope this helps. Since the file fails during loading, unfortunately `names()` will not work. Is there any way to update the attribute names via the header without loading the entire file?

**Reply by Jean-Romain** [reply]
Ok I see. Please report it as a bug.

**Reply by seanreilly66** [reply]
Issue created here https://github.com/r-lidar/lidR/issues/659

Thanks for all that you do. The package is incredible.
